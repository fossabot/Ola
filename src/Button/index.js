import React from 'react'
import { default as PT } from 'prop-types'
import cx from 'classnames'
import { getElementType } from '../utils'

import Spinner from '../Spinner'
import Icon from '../Icon'

const Button = ({ variant, children, disabled, busy, className, iconLink, as, ...props }) => {
  const ElementType = getElementType(Button, { as: as, ...props })
  delete props['as']
  const styles = cx(
    'ola_button',
    { [`is-${variant}`]: variant },
    { 'is-busy': busy },
    { 'is-disabled': disabled && !busy },
    className
  )
  const isIcon = (variant === 'link' && iconLink)
  return (
    <ElementType className={styles} disabled={busy ? true : disabled} {...props}>
      {busy && <Spinner />}
      <span className="ola_button-text">{!busy ? children : busy}</span>
      {isIcon && <Icon name="iconLink" />}
    </ElementType>
  )
}

Button.defaultProps = {
  as: 'button',
  variant: null,
  busy: null,
  className: null,
  disabled: false,
  iconLink: false
}

Button.propTypes = {
  /** Render Button with any html tag */
  as: PT.string,
  /** Button variants */
  variant: PT.oneOf(['primary', 'secondary', 'destructive-primary', 'destructive', 'pro', 'link']),
  /** Text for loading state */
  busy: PT.oneOfType([
    PT.string,
    PT.bool
  ]),
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired,
  /** Button is disabled */
  disabled: PT.bool,
  /**Icon is a Bolean  */
  iconLink: PT.bool
}

export default Button
