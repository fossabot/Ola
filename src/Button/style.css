.button {
    --color: var(--brand);
    --background: transparent;
    --color-hover: var(--brand-dark);
    --background-hover: var(--gray-xlight);
    --focus: var(--brand-focus);
    --background-active: var(--brand-focus);

    display: inline-flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    height: var(--size-8);
    min-width: var(--size-10);
    border: none;
    background: none;
    color: var(--color);
    background-color: var(--background);
    border-radius: calc(var(--size-8) / 2);
    font: var(--font-callout);
    padding: 0 1em;
    font-weight: var(--bold);
    transition-property: background-color, color, box-shadow;
    transition-duration: var(--duration);
    outline: 0;
    white-space: nowrap;
    text-decoration: none;
    box-sizing: border-box;

    &::-moz-focus-inner {
        border: 0;
    }

    &:hover {
        color: var(--color-hover);
        background-color: var(--background-hover);
    }

    &:focus {
        box-shadow: 0 0 0 4px var(--focus);
    }

    &:active {
        background-color: var(--background-active);
    }

    &.is-primary {
        --color: var(--white);
        --background: var(--brand);
        --color-hover: var(--white);
        --background-hover: var(--brand-dark);
        --focus: var(--brand-focus);
        --background-active: var(--black);
    }

    &.is-secondary {
        border: solid 1px var(--gray-light);

        &:hover {
            border-color: currentColor;
        }
    }

    &.is-destructive-primary {
        --color: var(--white);
        --background: var(--error);
        --color-hover: var(--white);
        --background-hover: var(--error-dark);
        --focus: var(--error-focus);
        --background-active: var(--black);
    }

    &.is-destructive {
        --color: var(--error);
        --focus: var(--error-focus);
        --color-hover: var(--error-dark);
        --background-active: var(--error-focus);

        border: solid 1px var(--gray-light);

        &:hover {
            border-color: currentColor;
        }
    }

    &.is-pro {
        --color: var(--white);
        --background: var(--pro);
        --color-hover: var(--white);
        --background-hover: var(--pro-dark);
        --focus: var(--pro-focus);
        --background-active: var(--black);
    }

    &.is-busy {
        color: var(--white);
        background-color: var(--gray);
        cursor: default;
        min-width: 0;
        padding-left: 0;
        cursor: wait;
    }

    &.is-link {
        --background-hover: transparent;

        cursor: pointer;
        min-width: 0;
        padding-left: 0;
        padding-right: 0;
        border-radius: 0;

        &:hover {
            text-decoration: underline;
        }
    }

    &:disabled {
        color: var(--gray-light);
        --background: var(--gray-xlight);
        --background-hover: var(--gray-xlight);
        --background-active: var(--gray-xlight);
        border: none;
        cursor: not-allowed;
        box-shadow: none;
    }
}

.button-group {
    display: flex;

    & .button + .button {
        margin-left: var(--size-6);
    }
}
