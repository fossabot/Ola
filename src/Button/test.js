import React from 'react'
import { Button } from './'
import renderer from 'react-test-renderer'

it('No variant button', () => {
  const tree = renderer
    .create(<Button>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Primary button', () => {
  const tree = renderer
    .create(<Button variant="primary">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Secondary button', () => {
  const tree = renderer
    .create(<Button variant="secondary">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('PRO button', () => {
  const tree = renderer
    .create(<Button variant="pro">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Primary destructive button', () => {
  const tree = renderer
    .create(<Button variant="destructive-primary">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Destructive button', () => {
  const tree = renderer
    .create(<Button variant="destructive">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Busy button', () => {
  const tree = renderer
    .create(<Button variant="primary" busy>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Primary button with extra class', () => {
  const tree = renderer
    .create(<Button variant="primary" extraClass="is-extra">Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Disabled primary button', () => {
  const tree = renderer
    .create(<Button variant="primary" disabled>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Disabled link button', () => {
  const tree = renderer
    .create(<Button variant="link" disabled>Click here</Button>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})