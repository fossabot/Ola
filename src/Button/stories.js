import React from 'react'
import Button from './'

export default {
  title: 'Button',
  component: Button,
  args: {
    children: 'Button',
  },
  argTypes: {
    onClick: { action: 'clicked' },
    onFocus: { action: 'focused' },
    busy: { control: 'text' },
    as: {
      control: {
        type: 'inline-radio',
        options: ['button', 'a', 'span'],
      },
    },
  },
}

export const Base = (args) => <Button {...args}>{args.children}</Button>
export const WithOlaIcon = (args) => (
  <Button variant="secondary" icon="add" {...args}>
    {args.children}
  </Button>
)
export const WithIcon = (args) => (
  <Button
    variant="secondary"
    icon={<img src="https://placeholder.pics/svg/20" />}
    {...args}
  >
    {args.children}
  </Button>
)
