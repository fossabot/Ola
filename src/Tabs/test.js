import React from 'react'
import Button from '../Button'
import TabSelector from './'
import TabContent from './Content'
import renderer from 'react-test-renderer'

it('Segment with two buttons', () => {
  const tree = renderer
    .create(
      <>
        <TabSelector>
          <Button variant="secondary" selected>Tab 1</Button>
          <Button variant="secondary">Tab 2</Button>
        </TabSelector>
        <TabContent selected>Tab content 1</TabContent>
        <TabContent>Tab content 2</TabContent>
      </>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
