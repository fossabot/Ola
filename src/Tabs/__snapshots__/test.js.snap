// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segment with two buttons 1`] = `
Array [
  <nav
    className="ola_tabs"
  >
    <button
      aria-pressed="true"
      className="ola_button is-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="ola_button-text"
      >
        Tab 1
      </span>
    </button>
    <button
      aria-pressed="false"
      className="ola_button is-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="ola_button-text"
      >
        Tab 2
      </span>
    </button>
    <div />
  </nav>,
  <div
    className="ola_tab selected"
    label="Tab 1"
  >
    Tab 1 content
  </div>,
  <div
    className="ola_tab"
    label="Tab 2"
  >
    Tab 2 content
  </div>,
]
`;
